db/migrate/20200202052500_create_communities_table.rb

```diff
@@ -0,0 +1,68 @@
+class CreateCommunitiesTable < ActiveRecord::Migration[5.2]
+  # helper User model without delegation for migration purposes
+  class MigrationSafeUser < ActiveRecord::Base; self.table_name = 'users'; end
+
+  def change
+    create_table :communities do |t|
+      t.string :name, null: false
+      t.string :host, null: false, index: true
+      t.timestamps
+    end
+
+    create_table :community_users do |t|
+      t.references :community, null: false, foreign_key: true, index: true
+      t.references :user, null: false, foreign_key: true, index: true
+      t.boolean :is_moderator
+      t.boolean :is_admin
+      t.integer :reputation
+      t.timestamps
+    end
+
+    reversible do |dir|
+      dir.up do
+        @community = Community.create(name: 'Sample', host: 'sample.qpixel.com')
+        community_users = MigrationSafeUser.pluck(:id, :is_admin, :is_moderator, :reputation).map do |user_id, is_admin, is_moderator, reputation|
+          {community_id: @community.id, user_id: user_id, is_admin: is_admin, is_moderator: is_moderator, reputation: reputation}
+        end
+        CommunityUser.create(community_users)
+        change_table :users do |t|
+          t.remove :is_moderator
+          t.remove :is_admin
```

Comment:
Moving is_admin and is_moderator is good because it allows for per community admins/mods - but we should keep an is_admin attribute on the global user model so that we can also have global admins. We'll also need to make some changes in views and controllers so that global admins can see and change anything, but community admins can only see and change things for their community.

Reply:
I think I'll modify that naming to is_global_moderator/is_global_admin.

Reply:
Works for me :+1: